app/controllers/users_controller.rb

```diff
@@ -26,10 +27,30 @@ def create
       render 'new'
     end
   end
+
+  def new_with_invitation_token 
+    invitation = Invitation.where(token: params[:token]).first
```

Comment:
You should use `find_by` for fetching single record rather than `where(...).first`.

HOMEWORK: Find out why it is better to use `find_by` method instead.